src/Servers/Kestrel/test/InMemory.FunctionalTests/BadHttpRequestTests.cs

```diff
@@ -48,17 +48,30 @@ public Task TestInvalidHeaders(string rawHeaders, string expectedExceptionMessag
                 expectedExceptionMessage);
         }
 
+        public static Dictionary<string, (string header, string errorMessage)> BadHeaderData => new Dictionary<string, (string, string)>
+        {
+            { "NullInHeaderName", ("Hea\0der: value", "Invalid characters in header name.") },
+            { "NullInHeaderValue", ("Header: va\0lue", "Malformed request: invalid headers.") },
```

Comment:
Does our `HttpParsingData` cause a similar issue? Or is this a problem only with `InlineData` not `MemberData`? If it's the former, we should also fix that. If it's the latter, we can simplify this to a single `IEnumerable<(string, string)>` property for the `MemberData` without changing the actual test.

https://github.com/dotnet/aspnetcore/blob/c39ecb9f9b387ec39112c3eace5ae38db06df160/src/Servers/Kestrel/shared/test/HttpParsingData.cs#L227-L234

Reply:
Nvm. `HttpParsingData` is escaped with our `EscapeNonPrintable` extension method further down in this test class. We should use `EscapeNonPrintable` to generate the key names like we do for `TestInvalidRequestLines` for consistency and to make it more obvious what the test data actually is without looking at the Dictionary.

https://github.com/dotnet/aspnetcore/blob/c39ecb9f9b387ec39112c3eace5ae38db06df160/src/Servers/Kestrel/test/InMemory.FunctionalTests/BadHttpRequestTests.cs#L237-L242

Reply:
It's the \x80 that's causing issues, \0 is fine. So the "MethodWithNullCharData" is safe.